src/main/java/org/medicmobile/webapp/mobile/SettingsDialogActivity.java

```diff
@@ -29,19 +29,22 @@
 public class SettingsDialogActivity extends LockableActivity {
 	private static final int STATE_LIST = 1;
 	private static final int STATE_FORM = 2;
-
 	private SettingsStore settings;
 	private ServerRepo serverRepo;
 	private int state;
 
 	private static class AppUrlVerificationTask extends ActivityBackgroundTask<SettingsDialogActivity, String, Void, AppUrlVerification> {
+
+		private AppUrlVerifier verifier = new AppUrlVerifier();
+
 		AppUrlVerificationTask(SettingsDialogActivity a) {
 			super(a);
 		}
 
 		protected AppUrlVerification doInBackground(String... appUrl) {
 			if(DEBUG && appUrl.length != 1) throw new IllegalArgumentException();
-			return new AppUrlVerifier().verify(appUrl[0]);
+			String chtUrl = verifier.clean(appUrl[0]);
```

Comment:
It feels like the `clean` functionality should just be built into the `verify` call (so that `verify` just calls `clean` internally and consumers of `AppUrlVerifier` do not have to call both methods).  Then `AppUrlVerifier.clean` can be a private method.

Seems like the only thing this really affects is that for the tests, you will have to assert like this:

```.js
assertEquals("https://example.com/uri", verifier.verify("https://example.com/uri").appUrl);
```
but I don't think that should be a problem.